src/goto-programs/goto_program.cpp

```diff
@@ -32,51 +37,80 @@ std::ostream& goto_programt::output_instruction(
   std::ostream& out,
   instructionst::const_iterator it) const
 {
-  out << "        // " << it->location_number << " ";
+  return output_instruction(ns, identifier, out, *it);
```

Comment:
Note that the iterator is passed by value, thus modifications to the iterator are not a risk. Yet the question as to whether we genuinely need both variants is a valid one.